packages/subgraph/src/packages/admin/adminMembers.ts

```diff
@@ -0,0 +1,63 @@
+import {dataSource, store} from '@graphprotocol/graph-ts';
+import {
+  Administrator,
+  AdministratorAdminPlugin
+} from '../../../generated/schema';
+import {Granted, Revoked} from '../../../generated/templates/DaoTemplate/DAO';
+
+export function handleGranted(event: Granted): void {
+  if (
+    isCorrectEvent(
+      event.params.permissionId.toHexString(),
+      event.params.where.toHexString()
+    )
+  ) {
+    let pluginAddress = event.params.where.toHexString();
+    let administratorAddress = event.params.who.toHexString();
+    let administrator = Administrator.load(administratorAddress);
+    if (!administrator) {
+      administrator = new Administrator(administratorAddress);
+      administrator.address = administratorAddress;
+      administrator.save();
+    }
+
+    let administratorMappingId = `${pluginAddress}_${administratorAddress}`;
+    let administratorPluginMapping = AdministratorAdminPlugin.load(
+      administratorMappingId
+    );
+    if (!administratorPluginMapping) {
+      administratorPluginMapping = new AdministratorAdminPlugin(
+        administratorMappingId
+      );
+      administratorPluginMapping.administrator = administratorAddress;
+      administratorPluginMapping.plugin = pluginAddress;
+      administratorPluginMapping.save();
+    }
+  }
+}
+
+export function handleRevoked(event: Revoked): void {
+  if (
+    isCorrectEvent(
+      event.params.permissionId.toHexString(),
+      event.params.where.toHexString()
+    )
+  ) {
+    let mappingId = `${event.params.where.toHexString()}_${event.params.who.toHexString()}`;
+    if (AdministratorAdminPlugin.load(mappingId)) {
+      store.remove('AdministratorAdminPlugin', mappingId);
+    }
+  }
+}
+
+function isCorrectEvent(permissionId: string, where: string): boolean {
+  let context = dataSource.context();
```

Comment:
```
let context = dataSource.context();
let requiredPermissionId = context.getString('permissionId');
let pluginAddress = context.getString('pluginAddress');
return permissionId == requiredPermissionId && where == pluginAddress;
```

Reply:
I'd rather have different ifs to not impact the indexing speed of subgraph too much

Reply:
so, `context.getString('pluginAddress');` would impact the speed, right ?

Reply:
right because it needs to load things